### PR TITLE
chore(ui): remove references to bootstrap

### PR DESCRIPTION
**Pull request by EFF**
Ok this one seems like a lot but in fact it's not much except removing all reference to reactstrap and react-bootstrap component and replace them with either blueprint components or our own.

**Note: I didn't remove the react-bootstrap library because some modules still depend on it and maybe some users built their own ui on top of react-bootstrap. I'll survey our customer success team to see if our some enterprise clients built something that depends on it and eventually add a deprecation notice for others**

At the same time, I converted old jsx files to tsx files and also removed all unused components.

The reason I did this is mainly because I plan on improving the studio a lot in the upcoming months and I wanted to fully take ownership of the code, starting by exploring the oldest stuff helps to create a good mental map.

GLHF 🤷

### CODE REVIEW

File: packages/studio-ui/src/web/views/FlowBuilder/common/actionPopover.tsx [hunk 1]
```diff
@@ -13,14 +12,11 @@ interface Props {
...
-  const [target, setTarget] = useState<HTMLElement>(null)
-
   const actionInstruction = parseActionInstruction(props.text.trim())
-  const actionName = `${actionInstruction.actionName} (args)`
+  const actionName = `${actionInstruction.actionName}`
 
-  let callPreview: string
-  if (actionInstruction.argsStr) {
+  let callPreview: string = ''
+  if (actionInstruction.argsStr && actionInstruction.argsStr !== '{}') {
```
Comment (laurentlp) [inline packages/studio-ui/src/web/views/FlowBuilder/common/actionPopover.tsx:19]: Can you please move `'{}'` into a constant

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if (text) {
+      const alreadyExists = this.props.flow.nodes.find((x) => x.name === text)
+      if (!alreadyExists) {
+        this.props.updateNode({ name: text })
+      }
+    }
+  }
+
+  transformText(text) {
+    return text.replace(/[^a-z0-9-_\.]/gi, '_')
```
Comment (laurentlp) [inline packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx:32]: Can you either move the regex into a constant or add a comment that explains what this regex does?

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          <Tag minimal>{node.next?.length ?? 0}</Tag> {lang.tr('studio.flow.node.transitions')}
+        </H3>
+        <TransitionSection
+          readOnly={readOnly}
+          items={node.next}
+          currentFlow={this.props.flow}
+          currentNodeName={this.props.node.name}
+          subflows={this.props.subflows}
+          onItemsUpdated={(items) => this.props.updateNode({ next: items })}
+          //missing props
```
Comment (laurentlp) [inline packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx:64]: What does it mean?

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    props.updateNode({ name: text })
+  }
+
+  return (
+    <div className={style.node}>
+      <EditableInput
+        /* We should always sugest that the name should be changed
+             if the node has the default name and it is the last created */
```
Comment (laurentlp) [inline packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx:88]: Is this a TODO?

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        key={node.id}
+        shouldFocus={isLastNode && node.name.match(/node-(\w|\n){4}$/g)}
```
Comment (laurentlp) [inline packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx:90]: Can you move the regex to a const.

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+interface Props {
+  flow: FlowView
+  node: NodeView
+  subflows: string[]
+  readOnly: boolean
+  isLastNode: boolean
+  updateNode: (node: Partial<NodeView>) => void
+  copyFlowNodeElement: Function
+  pasteFlowNodeElement: Function
+  buffer: any
```
Comment (laurentlp) [inline packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx:32]: Can you try to improve those typings?

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx (hunk 2 above)
Comment (EFF) [reply packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx:32]: I dnd't touch this piece of code yet, I'll eventually get into reviewing this, will fix only then

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx (hunk 3 above)
Comment (EFF) [reply packages/studio-ui/src/web/views/FlowBuilder/nodeProps/SkillCallNode.tsx:64]: fixed, added better comment

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx (hunk 4 above)
Comment (EFF) [reply packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx:88]: Didn't write this, I think it's a leftover from the old days, I'll leave it there

File: packages/studio-ui/src/web/views/FlowBuilder/common/TransitionItem.tsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    const { position } = this.props
+    const { condition, caption } = this.props.transition
+
+    if (caption) {
+      const vars = {}
+      const htmlTpl = caption.replace(/\[(.+)]/gi, (x) => {
+        const name = stripDots(x.replace(/[\[\]]/g, ''))
+        vars[name] = `<span class="val">${_.escape(name)}</span>`
+        return '{{{' + name + '}}}'
+      })
```
[the remark this answers is not in the document]
Comment (EFF) [reply packages/studio-ui/src/web/views/FlowBuilder/common/TransitionItem.tsx:75]: I'll completely review `ActionItem` and `TransitionItem` in another pull request=

File: packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx (hunk 6 above)
Comment (EFF) [reply packages/studio-ui/src/web/views/FlowBuilder/nodeProps/StandardNode.tsx:32]: I have no idea what they are at the moment but they are used everywhere, I'll try to fix

**Review by laurentlp — APPROVED** [review]

**Pull request merged**